packages/adapter-d1/src/index.ts

```diff
@@ -0,0 +1,353 @@
+}
+
+export async function deleteRecord(db:BothDB, SQL:string, bindings:any[]) {
+  // eslint-disable-next-line no-useless-catch
+  try {
+    bindings = cleanBindings(bindings)
+    await db.prepare(SQL).bind(...bindings).run()
+  } catch(e:any) {
+    console.log(`Error deleting record with ${SQL} and bindings ${bindings}`)
+    console.log(`D1 ERROR MESSAGE`, e.message, e.cause?.message)
+    throw e
+  }
+}
+
+/**
+ *
+ * ## Setup
+ *
+ * This is the D1 Adapter for [`next-auth`](https://authjs.dev). This package can only be used in conjunction with the primary `next-auth` package. It is not a standalone package.
+ *
+ * ### Configure Auth.js
+ *
+ * ```javascript title="pages/api/auth/[...nextauth].js"
+ * import NextAuth from "next-auth"
+ * import { D1Adapter, up } from "@next-auth/d1-adapter"
+ *
+ *
+ * // For more information on each option (and a full list of options) go to
+ * // https://authjs.dev/reference/configuration/auth-options
+ * export default NextAuth({
+ *   // https://authjs.dev/reference/providers/
+ *   providers: [],
+ *   adapter: D1Adapter(env.db)
+ *   ...
+ * })
+ * ```
+ *
+ * ### Migrations
+ *
+ * Somewhere in the initialization of your application you need to run the `up(env.db)` function to create the tables in D1.
+ * It will create 4 tables if they don't already exist:
+ * `authjs_accounts`, `authjs_sessions`, `authjs_users`, `authjs_verification_tokens`.
+ * 
+ * The table prefix "authjs_" is not configurable at this time.
+ * 
+ * You can use something like the following to attempt the migration once each time your worker starts up.  Running migrations more than once will not erase your existing tables.
+ * ```javascript
+ * import { up } from "@next-auth/d1-adapter"
+ * 
+ * let migrated = false;
+ * async function migrationHandle({event, resolve}) {
+ *  if(!migrated) {
+ *    try {
+ *      await up(event.platform.env.db)
+ *      migrated = true
+ *    } catch(e) {
+ *      console.log(e.cause.message, e.message)
+ *    }
+ *  }
+ *  return resolve(event)
+ * }
+ * ```
+ * 
+ *
+ * You can also initialize your tables manually.  Look in [init.ts](https://github.com/nextauthjs/next-auth/packages/adapter-d1/src/migrations/init.ts) for the relevant sql.
+ * Paste and run the SQL into your D1 dashboard query tool.
+ * 
+ **/
+export function D1Adapter(db:BothDB): Adapter {
+
+  // we need to run migrations if we dont have the right tables
+
+  return {
+    async createUser(user) {
+      console.log('Creating User:', user)
+      const id: string = crypto.randomUUID();
+      const createBindings = [id, user.name , user.email, user.emailVerified?.toISOString(), user.image];
+      const getBindings = [id];
+      const newUser = await createRecord<AdapterUser>(
+        db, 
+        CREATE_USER_SQL, createBindings,
+        GET_USER_BY_ID_SQL, getBindings
+      );
+      // this method cant return null, gotta throw
+      if(newUser) return newUser;
+      throw new Error(`Couldn't create a new user`);
```

Comment:
createRecord is throwing, too, so I think we can remove this line? Same for session